### ISSUE TITLE
Context deadline exceeded

### ISSUE THREAD

**Issue by tungpt00179** [minio/console#1135]
Hi guys,

I've updated minio docker image to lastest this morning, according to documentation, I changed env variables, and the API work fine, i can access server with mc client, but when I access to console via browser, it shows invalid login with a message: "**Context deadline exceeded**"
![Screen Shot 2021-10-21 at 12 24 00](https://user-images.githubusercontent.com/12286322/138217267-a4aed8ff-ce10-45f9-b9ed-743038eb4b74.png)

I tried to trace the server with command: **mc admin trace server-dump-online -a -v** and there's a result log:
```
[REQUEST sts.AssumeRole] [2021-10-21T05:26:11:000] [Client IP:xxxx]
server.dump.online POST /
server.dump.online Proto: HTTP/1.1
server.dump.online Host: server.dump.online
server.dump.online Forwarded: for=x.x.x.x;host=server.dump.online;proto=https;proto-version=
server.dump.online User-Agent: Go-http-client/1.1
server.dump.online X-Amz-Date: 20211021T052611Z
server.dump.online X-Forwarded-Port: 443
server.dump.online Authorization: AWS4-HMAC-SH...A256 Credential=WJAxsK./20211021//sts/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=8c274....
server.dump.online Content-Type: application/x-www-form-urlencoded
server.dump.online X-Forwarded-For: x.x.x.x
server.dump.online X-Forwarded-Host: server.dump.online
server.dump.online X-Forwarded-Proto: https
server.dump.online Content-Length: 57
server.dump.online Action=AssumeRole&DurationSeconds=3600&Version=2011-06-15
server.dump.online [RESPONSE] [2021-10-21T05:26:11:000] [ Duration 27.076ms  ↑ 204 B  ↓ 1.1 KiB ]
server.dump.online 200 OK
server.dump.online Content-Type: application/xml
server.dump.online Server: MinIO
server.dump.online X-Amz-Request-Id: 16AFF474924988C1
server.dump.online X-Content-Type-Options: nosniff
server.dump.online X-Xss-Protection: 1; mode=block
server.dump.online Accept-Ranges: bytes
server.dump.online Content-Length: 772
server.dump.online Content-Security-Policy: block-all-mixed-content
server.dump.online Strict-Transport-Security: max-age=31536000; includeSubDomains
server.dump.online Vary: Origin
server.dump.online <?xml version="1.0" encoding="UTF-8"?>
<AssumeRoleResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/"><AssumeRoleResult><AssumedRoleUser><Arn></Arn><AssumeRoleId></AssumeRoleId></AssumedRoleUser><Credentials><AccessKeyId>xxxxxxxx</AccessKeyId><SecretAccessKey>hPRr....</SecretAccessKey><Expiration>2021-10-21T06:26:11Z</Expiration><SessionToken>eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.....</SessionToken></Credentials></AssumeRoleResult><ResponseMetadata><RequestId>16AFF474924988C1</RequestId></ResponseMetadata></AssumeRoleResponse>
```
So what's wrong with my configuration?
i run on k8s with following env variables:

MINIO_BROWSER_REDIRECT_URL=console-domain
MINIO_SERVER_URL=api-domain
CONSOLE_MINIO_SERVER=internal-minio.svc.local:9000

Thanks

**Comment by harshavardhana** [comment]
Your server URL perhaps is not resolvable.

**Comment by dvaldivia** [comment]
Is this still a problem with the latest builds?

**Comment by tungpt00179** [comment]
@dvaldivia I try with the latest build but the error still occurs, as you can see below.
All my domains can be accessed from the internet.

![Screen Shot 2021-11-19 at 11 48 22](https://user-images.githubusercontent.com/12286322/142566694-f2204923-dacf-4a51-8f5b-3fad531ef975.png)

**Comment by tungpt00179** [comment]
for more information, i deploy seperate minio/console docker image and this is log from node app:


I: 2021/11/19 12:22:46 Serving console at http://[::]:9090
--
  | I: 2021/11/19 12:23:03 Unable to validate the session token AMLxzRU7Zswi8jSNzeJ3Mhfns3NywsY0wWeMj/I5zFlqCfvSwIMDBh8lFbj97hg0+UPjXcM2QUsMm7NArp7XM+MVbbUdqsExgCdvkyQd3CBp2Rjg815MixaN8jwmrPwZ+ZzmDg77eVq88RAy5h/jLnIliM2EOTc+/Co3tPQ38oVK4zb8UbcfgaeRClqqvqEmz8V0ZFrS3SPIyCfEahVJuO8MKH8CT2zoqe0bgWLed8Mxm3sQEE671IGYEfo/e5DGB4gtzWHoHP6pawovR9Wvtq/1zoKhqInFSHKNmaOqueEuJZ/j4K5MYOn1VQyGcXS0F9igg9lagLUl4oCB3qTxO3PIL4aDlPCrvZrBmsdhwV1HUFIwQu/VzT1vKYmZV448SExeWvGSF+gVfMW39O9WN0Y9319BalLNYIAl4u11eMxrRhofFDS/RMsBH7pXuMFIlWwtP7Aa9Cbvr6tmMMZ9WlR17QHMJuVlwq249qUC1u/dBJR2WM47l5nR4kgq6Yl5BRjPoEJ5Q+yhIgI/rHKjWrjd3FQyfdE5cdSb5+F3G3yoDfsht+n8H2xfYl/r0sWedILMDq48bWub55xS78NPpB3jXuE=: session token internal data is malformed
  | E: 2021/11/19 12:23:31 original error -> (user_session.go:61: context deadline exceeded)
  | E: 2021/11/19 12:23:31 friendly error: invalid session

**Comment by mengqian277** [comment]
This problem occurs when creating a certificate using OpenSSL

**Comment by momiji** [comment]
Hello, I have the same issue.

Deployed latest release of minio and separate console, from docker hub also.

I've done a lot of tests and I believe the reason is related to:
- the timeout is set to 20 seconds, looks like it calls AdminInfo() or something like that
- the user has a lot of policies, which may explains why it is so long

After further analysis, the impacted user has :
- 2 groups with policies defined for each group
- 57 policies, each of them containing 5 statements
- 225 unique statements (instead of 285=57*5), as some of them are defined in multiple policies

I made a rapid test by creating 3 policies to replace them all (only 1 is too big > 50KB ? - I don't know what is the limit, I had to split in 0-100 + 100-200 + 200-225), to see if the issue still occurs. This is no more the case, the answer came in 1s.

I don't know if technically there is a way to speed up the AdminInfo() call when there are a lot of policies. Maybe parallel loading of policies?
The circumvention is to denormalize the policies to have a very small number of policies per user. This is a pain of course, as one need to generate denormalized policies from existing groups and so on, but it is still doeable.

Does it help?
